newrelic/hooks/mlmodel_openai.py

```diff
@@ -192,18 +191,38 @@ def wrap_chat_completion_create(wrapped, instance, args, kwargs):
         trace_id = available_metadata.get("trace.id", "")
 
         try:
-            response = wrapped(*args, **kwargs)
+            return_val = wrapped(*args, **kwargs)
         except Exception as exc:
-            exc_organization = getattr(exc, "organization", "")
+            if OPENAI_V1:
+                response = getattr(exc, "response", "")
+                response_headers = getattr(response, "headers", "")
+                exc_organization = response_headers.get("openai-organization", "") if response_headers else ""
+                # There appears to be a bug here in openai v1 where despite having code,
+                # param, etc in the error response, they are not populated on the exception
+                # object so grab them from the response object instead.
+                content = getattr(response, "content", b"{}")
+                response = json_decode(content.decode("utf-8")).get("error", {})
```

Comment:
It looks like the status code and other notice error attributes may be available on `exc.body` `exc.message`.

Reply:
I've changed it to use the body. The exc.message is not the message we are looking for. 😂 It includes the json inside of it which is the same as the default message that we would grab without overriding it so I'm using the message from the body instead still.